File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

import entity.PageResult;
import entity.Result;

@RestController
@RequestMapping("/brand")
public class BrandController {

	@Reference
	private BrandService brandService;
	
	@RequestMapping("/findAll")
	public List<TbBrand> findAll(){
		
		return brandService.findAll();
	}
	
	/**
	 * 这是分页查询的方法
	 * @param page 当前页
	 * @param rows 每页记录数
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult findPage(int page,int rows){
		
		return brandService.findPage(page, rows);
	}

	/**
	 * 这是添加品牌的方法
	 * @param brand
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody TbBrand brand){
		
		try {
			brandService.add(brand);
			return new Result(true, "品牌添加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "品牌添加失败");
		}
				
	}
	/**
	 * 这是查询单一一个品牌的方法
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public TbBrand findOne(Long id){
		
		return brandService.findOne(id);
	}
	
	
	@RequestMapping("/update")
	public Result update(@RequestBody TbBrand brand){
		
		try {
			brandService.update(brand);
			return new Result(true, "品牌修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "品牌修改失败");
		}
				
	}
		
	@RequestMapping("/delete")
	public Result delete(Long[] ids){
		
		try {
			brandService.delete(ids);;
			return new Result(true, "品牌删除成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "品牌删除失败");
		}
	}
	
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbBrand brand,int page,int rows){
		
		return brandService.findPage(brand, page, rows);
	}
	
	@RequestMapping("/selectOptionList")
	public List<Map> selectOptionList(){
		
		return brandService.selectOptionList();
	}
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController',function($scope,$controller,brandService){
			
			$controller('baseController',{$scope:$scope});
	
			$scope.findAll=function(){
				brandService.findAll().success(
						function(response){
							$scope.list=response
							
				})
				
			}
			
			$scope.findPage = function(page,rows){
				brandService.findPage(page,rows).success(
					function(response){
						$scope.list=response.rows;
						$scope.paginationConf.totalItems=response.total;
					}		
				)
				
			}
			
			$scope.save=function(){
				var object=null;
				if($scope.entity.id!=null){
					object=brandService.update($scope.entity)
				}else{
					object=brandService.add($scope.entity)
				}
				object.success(
					function(response){
						if(response.success){
							$scope.reloadList();
						}else{
							alert(response.message);
						}
					}		
				)
				
			}
			
			$scope.findOne=function(id){
				brandService.findOne(id).success(
					function(response){
						$scope.entity=response
					}		
				)
			}
			
			
			
			$scope.delete=function(){
				brandService.delete($scope.selectIds).success(
					function(response){
						if(response.success){
							$scope.reloadList();
						}else{
							alert(response.message);
						}
					}		
				)
				
			}
			
			$scope.searchEntity={};
			$scope.search=function(page,rows){
				brandService.search(page,rows,$scope.searchEntity).success(
						function(response){
							$scope.list=response.rows;
							$scope.paginationConf.totalItems=response.total;
						}		
				)
				
			}
			
		})